libevm/VM.cpp

```diff
@@ -543,6 +544,7 @@ void VM::interpretCases()
 
 		CASE_BEGIN(BALANCE)
 		{
+			m_runGas += toUint64(m_schedule->balanceGas);
```

Comment:
You should change the schedule itself, not just add something to rungas here manually.